test/jdk/java/awt/dnd/RejectDragDropActionTest.java

```diff
@@ -0,0 +1,122 @@
+import java.awt.Frame;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.datatransfer.StringSelection;
+import java.awt.dnd.DnDConstants;
+import java.awt.dnd.DragGestureListener;
+import java.awt.dnd.DragGestureRecognizer;
+import java.awt.dnd.DragSource;
+import java.awt.dnd.DropTarget;
+import java.awt.dnd.DropTargetAdapter;
+import java.awt.dnd.DropTargetDragEvent;
+import java.awt.dnd.DropTargetDropEvent;
+import java.awt.dnd.DropTargetListener;
+import java.awt.event.InputEvent;
+import java.lang.reflect.InvocationTargetException;
+
+
+public class RejectDragDropActionTest {
+
+    private static boolean incorrectActionDetected = false;
+
+    private static final int FRAME_ACTIVATION_TIMEOUT = 3000;
+
+    private static Frame frame;
+    private static DragSource ds;
+    private static DragGestureListener dgl;
+    private static DragGestureRecognizer dgr;
+    private final DropTargetListener dtl = new DropTargetAdapter() {
+            public void dragEnter(DropTargetDragEvent dtde) {
+                dtde.rejectDrag();
+            }
+            public void dragOver(DropTargetDragEvent dtde) {
+                if (dtde.getDropAction() == DnDConstants.ACTION_NONE) {
+                    incorrectActionDetected = true;
+                }
+            }
+            public void drop(DropTargetDropEvent dtde) {
+                dtde.rejectDrop();
+            }
+        };
+    private final DropTarget dt = new DropTarget(frame, dtl);
+
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException, AWTException {
+        EventQueue.invokeAndWait(() -> {
+            frame = new Frame("RejectDragDropActionTest");
+            ds = DragSource.getDefaultDragSource();
+            dgl = dge -> dge.startDrag(null, new StringSelection("OOKK"));
+            dgr = ds.createDefaultDragGestureRecognizer(frame, DnDConstants.ACTION_COPY, dgl);
+            frame.setBounds(100, 100, 200, 200);
+            frame.setVisible(true);
+        });
+
+        try {
+            Robot robot = new Robot();
+            robot.waitForIdle();
+            robot.delay(FRAME_ACTIVATION_TIMEOUT);
+
+            Point startPoint = frame.getLocationOnScreen();
+            Point endPoint = new Point(startPoint);
+            startPoint.translate(50, 50);
+            endPoint.translate(150, 150);
+
+            robot.mouseMove(startPoint.x, startPoint.y);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            for (Point p = new Point(startPoint); !p.equals(endPoint);
+                 p.translate(sign(endPoint.x - p.x),
+                         sign(endPoint.y - p.y))) {
+                robot.mouseMove(p.x, p.y);
+                robot.delay(50);
+            }
+
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+        } finally {
+            if (frame != null) {
+                frame.dispose();
+            }
+        }
+
+        if (incorrectActionDetected) {
```

Comment:
incorrectActionDetected is set on one thread an used on another, same for "frame" in other tests.

Reply:
meaning you are suggesting they need to be volatile ?

Reply:
or changed to be used on the same thread, in this case EDT. It seems different tests use different approaches.